splunk-otel-android/src/main/java/com/splunk/rum/PostApi29NetworkDetector.java

```diff
@@ -46,24 +46,30 @@
     @SuppressLint("MissingPermission")
     @Override
     public CurrentNetwork detectCurrentNetwork() {
-        NetworkCapabilities capabilities = connectivityManager.getNetworkCapabilities(connectivityManager.getActiveNetwork());
-        if (capabilities == null) {
-            return NO_NETWORK;
-        }
-        String subType = null;
-        if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)) {
-            // If the app has the permission, use it to get a subtype.
-            if (hasPermission(Manifest.permission.READ_PHONE_STATE)) {
-                subType = getDataNetworkTypeName(telephonyManager.getDataNetworkType());
+        try {
+            NetworkCapabilities capabilities = connectivityManager.getNetworkCapabilities(connectivityManager.getActiveNetwork());
+            if (capabilities == null) {
+                return NO_NETWORK;
+            }
+            String subType = null;
```

Comment:
Unrelated to this change, but looks like the `subType` should be moved into the `if` scope block.